pkg/core/label_list.go

```diff
@@ -133,7 +139,23 @@ func listLabelsChan(repo string, stores context2.Stores, prefix string, opts ...
 	keysChan := make(chan keyBatchEvent, 1)
 
 	iterator := func(next string) ([]string, string, error) {
-		return GetLabelStore(stores).KeysPrefix(context.Background(), next, model.GetArchivePathPrefixToLabels(repo, prefix), "", settings.batchSize)
+		if settings.label != "" {
+			return GetLabelStore(stores).KeysPrefix(context.Background(), next,
+				model.GetArchivePathPrefixToLabels(repo, prefix),
+				"", settings.batchSize)
+		} else {
+			// tbd: sandbox o.(type).method() type assert fail compared to o.(type).method() error
+			versionedStore := GetLabelStore(stores).(storage.StoreVersioned)
+			versions, versionsErr := versionedStore.KeyVersions(context.Background(), settings.label)
+			if versionsErr != nil {
+				return nil, "", versionsErr
```

Comment:
```suggestion
				return nil, "", fmt.Errorf("a backend store that supports versionning is required when using this option: %w" versionsErr)
```

Reply:
pretty sure there's a better way to chain errors than this ;)